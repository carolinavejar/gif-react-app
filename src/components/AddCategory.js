import React, { useState } from 'react'

export const AddCategory = () => {
    const  [inputValue, setInputValue] = useState('Hola');

    const handleInputChange = (e)=> {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e)=> {
        e.preventDefault();
        console.log(` * * *    handleSubmit    * * * `);;
    }
    return (
        <form onSubmit = { handleSubmit }>
            <input 
                type = "text"
                value =  { inputValue }
                className = "input-field"
                onChange = { handleInputChange }
            />
        </form>
    )
}
