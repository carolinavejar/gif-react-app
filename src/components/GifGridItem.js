import React from 'react'
import { shallow } from 'enzyme/build';

import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

export const GifGridItem = ( {id, title, url} ) => {
    return (
        <div className = "card animate__animated animate__bounce animate__delay-2s">
            <div className = "card-image">
                <img src= { url} alt = {title}></img> 
                <div className = "card-title">{title} </div>
            </div>
        </div>
    )
}
