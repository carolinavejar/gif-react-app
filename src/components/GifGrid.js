import React , { useState, useEffect } from 'react';
import axios from 'axios';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, []);
    // ejecuta el cÃ³digo de arriba solo la primera vez que se renderiza

    const getGifs = () => {
        const api_key = 'URLjL49KyzdscasPa5146vq3CmpIYQjq';
        const cat = 'RIHANNA';
        axios.get(`https://api.giphy.com/v1/gifs/search?api_key=${ api_key }&q=${ cat }&limit=10`)
        .then((response) => {
            const { data : { data } } = response;
            const gifs = data.map ( img => {
                return {
                    id: img.id,
                    title: img.title,
                    url: img.images.downsized_medium.url
                }
            });
            setImages(gifs);
        })
    };

    return (
        <>
            {
                images.map( (img) => 
                    <GifGridItem 
                        key = { img.id }
                        { ...img }
                    >

                    </GifGridItem>
                )
            }
        </>
    )
}
