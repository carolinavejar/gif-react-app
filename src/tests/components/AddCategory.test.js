import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"
import React from 'react';

describe(' Pruebas en componente <AddCategory',()=> {
    const setCategories = () => {};
    const wrapper = shallow(<AddCategory setCategories = {setCategories} />);

    test('debe mostrarse correctanmente', ()=> {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe cambiar input', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo'
        input.simulate('change', { target : {value : value}});

        expect(wrapper.find('p').text()).toBe(value);
    })
})