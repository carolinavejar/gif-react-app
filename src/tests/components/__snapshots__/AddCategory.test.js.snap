// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas en componente <AddCategory debe mostrarse correctanmente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    className="input-field"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
